docs/pages/deepSleepMgr_L2_Low-Level_TestSpecification.md

```diff
@@ -0,0 +1,128 @@
+# DEEPSLEEPMGR L2 Low Level Test Specification and Procedure Documentation
+
+## Table of Contents
+
+- [DEEPSLEEPMGR L2 Low Level Test Specification and Procedure Documentation](#deepsleepmgr-l2-low-level-test-specification-and-procedure-documentation)
+
+  - [Table of Contents](#table-of-contents)
+  - [Overview](#overview)
+    - [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+    - [Definitions](#definitions)
+    - [References](#references)
+  - [Level 2 Test Procedure](#level-2-test-procedure)
+
+## Overview
+
+This document describes the level 2 testing suite for the DEEPSLEEPMGR module.
+
+### Acronyms, Terms and Abbreviations
+
+- `HAL` \- Hardware Abstraction Layer, may include some common components
+- `UT`  \- Unit Test(s)
+- `OEM`  \- Original Equipment Manufacture
+- `SoC`  \- System on a Chip
+
+### Definitions
+
+  - `ut-core` \- Common Testing Framework <https://github.com/rdkcentral/ut-core>, which wraps a open-source framework that can be expanded to the requirements for future framework.
+
+### References
+- `High Level Test Specification` - [deepsleep-manager_TestSpec.md](deepsleep-manager_TestSpec.md)
+
+## Level 2 Test Procedure
+
+The following functions are expecting to test the module operates correctly.
+
+### Test 1
+
+|Title|Details|
+|--|--|
+|Function Name|`test_l2_deepSleepMgr_SetDeepSleepAndVerifyWakeup1`|
+|Description|Set the deep sleep with a duration of one second and verify the wake-up source|
+|Test Group|Module : 02|
+|Test Case ID|001|
+|Priority|High|
+
+**Pre-Conditions :**
+None
+
+**Dependencies :**
+None
+
+**User Interaction :**
+If user chose to run the test in interactive mode, then the test case has to be selected via console.
+
+#### Test Procedure :
+
+| Variation / Steps | Description | Test Data | Expected Result | Notes|
+| -- | --------- | ---------- | -------------- | ----- |
+| 01 | Initialize the deep sleep manager using PLAT_DS_INIT | None | DEEPSLEEPMGR_SUCCESS | Should be successful |
+| 02 | Set the deep sleep with a duration of one second and enable GPIO wakeup using PLAT_DS_SetDeepSleep | deep_sleep_timeout=1, isGPIOWakeup=TRUE, networkStandby=FALSE | DEEPSLEEPMGR_SUCCESS | Should be successful |
```

Comment:
isGPIOWakeup is an output parameter so it should have a valid pointer and we can set networkStanby always false in L2 test case as we don't depend on external networks to wake up.

Also on timeout, this variable should be set to "False" as per the interface file which says "TRUE for user related wake up, false for not"

Reply:
Done